zephyr/src/logging/spotflow_log_backend.c

```diff
@@ -0,0 +1,193 @@
+	size_t dropped_backend_count;
+	size_t message_index;
+};
+
+static struct spotflow_log_context spotflow_log_ctx;
+
+static void init(const struct log_backend* backend);
+
+static void process(const struct log_backend* backend, union log_msg_generic* msg);
+
+static void panic(const struct log_backend* backend);
+
+static void dropped(const struct log_backend* backend, uint32_t cnt);
+
+static const struct log_backend_api log_backend_spotflow_api = {
+	.init = init,
+	.process = process,
+	.panic = panic,
+	.dropped = dropped,
+	/* .sync_string and .sync_hexdump can be added if you use IMMEDIATE mode */
+};
+
+#ifdef CONFIG_LOG_BACKEND_SPOTFLOW
+/* can be autostarted because there is message queue where messages are stored */
+LOG_BACKEND_DEFINE(log_backend_spotflow, log_backend_spotflow_api, true /* autostart */,
+		   &spotflow_log_ctx);
+#endif /* CONFIG_LOG_BACKEND_SPOTFLOW */
+
+static int enqueue_log_msg(const struct spotflow_mqtt_msg* msg, struct spotflow_log_context* ctx);
+
+static void process_single_message_stats_update(struct spotflow_log_context* context, bool dropped);
+
+static void process_message_stats_update(struct spotflow_log_context* context, uint32_t cnt,
+					 bool dropped);
+
+static void init(const struct log_backend* const backend)
+{
+	LOG_INF("Spotflow log backend initialized.");
+	__ASSERT(backend->cb->ctx != NULL, "Spotflow log backend context is NULL");
+	struct spotflow_log_context* ctx = backend->cb->ctx;
+	int rc = spotflow_cbor_output_context_init(&ctx->cbor_output_context);
+	__ASSERT(result == 0, "Failed to initialize spotflow output context");
+	if (rc < 0) {
+		LOG_ERR("Failed to initialize spotflow output context: %d", rc);
+		return;
+	}
+	ctx->dropped_backend_count = 0;
+	ctx->message_index = 0;
+	spotflow_start_mqtt();
+}
+
+static void process(const struct log_backend* const backend, union log_msg_generic* msg)
+{
+	struct log_msg* log_msg = &msg->log;
+	struct spotflow_log_context* ctx = backend->cb->ctx;
+	if (ctx == NULL) {
+		LOG_DBG("No spotflow log context");
+		return;
+	}
+
+	uint8_t* cbor_data = NULL;
+	size_t cbor_data_len = 0;
+	int rc = spotflow_cbor_encode_message(log_msg, ctx->cbor_output_context, &cbor_data,
+					      &cbor_data_len);
+
+	if (rc < 0) {
+		LOG_DBG("Failed to encode message: %d", rc);
+		return;
+	}
+
+	/* Allocate memory for the message structure */
+	struct spotflow_mqtt_msg* mqtt_msg = k_malloc(sizeof(struct spotflow_mqtt_msg));
+	if (!mqtt_msg) {
+		LOG_DBG("Failed to allocate memory for message");
+		k_free(cbor_data);
+		return;
+	}
+
+	/* Set up the message */
+	mqtt_msg->payload = cbor_data;
+	mqtt_msg->len = cbor_data_len;
+
+	/* Enqueue the message (passing pointer) */
+	if (enqueue_log_msg(mqtt_msg, ctx) < 0) {
+		LOG_DBG("Unable to put message in queue, dropping");
+		k_free(mqtt_msg->payload);
+		k_free(mqtt_msg);
+		process_single_message_stats_update(ctx, true);
+	} else {
+		process_single_message_stats_update(ctx, false);
```

Comment:
```suggestion
		process_single_message_stats_update(ctx, true /* dropped */);
	} else {
		process_single_message_stats_update(ctx, false /* dropped */);
```

Or use an enum instead of `bool`.